kernel/thread/thread.c

```diff
@@ -327,6 +342,82 @@ int thd_remove_from_runnable(kthread_t *thd) {
     return 0;
 }
 
+/* Creates and initializes the static TLS segment for a thread,
+   composed of a Thread Control Block (TCB), followed by .TDATA,
+   followed by .TBSS, very carefully ensuring alignment of each
+   subchunk. 
+*/
+static void* thd_create_tls_data(void) {
```

Comment:
Please move all variables to the top of the function. In-kernel we stick with C89/C90-style.

Also, please replace aligned_alloc with memalign since we don't compile the main kernel as C11. `<malloc.h>` is already included, so the function should be prototyped already.